Improve woo content styles in block email

costasovo opened the pull request:
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- If necessary, indicate if this PR is part of a bigger feature. Add a label with the format `focus: name of the feature [team:name of the team]`. -->

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

<!-- For bug fixes: If known, please provide links to help with traceability and escape analysis. -->
<!-- Please include a link to the issue of the bug being fixed, if one doesn't exist please create it. -->
<!-- If the PR that introduced the bug is known, please also add its link below. -->

Closes #56020


### Screenshots or screen recordings:

<!-- If this PR includes UI changes, please provide screenshots or a screen recording for clarity. -->
<!-- This section can be removed if not applicable. -->

Note: I used the color to highlight applied email editor styles.

| Before | After |
| ------ | ----- |
|     ![Screenshot 2025-03-07 at 11 16 06](https://github.com/user-attachments/assets/3a37ed63-4260-4a99-b6fb-6ba8a27fe8b0) |    ![Screenshot 2025-03-07 at 11 08 13](https://github.com/user-attachments/assets/30fb17f2-e10d-49ae-8660-547c6bff9cfb)   |


## Changes
1. **Woo Commerce Captured Content Styling Enhancement**
   - Added inlining of styles configured in the block editor
   - Removed a couple of styles (color, font-family) inlined from Woo. As for now, this looks sufficient. 

2. **Woo Content Capture Refactoring**
   - Moved mail content capture logic to `WooContentProcessor`
   - Improves separation of concerns and code organization and we get reusable serivice for capturing the content


<!-- Begin testing instructions -->

### How to test the changes in this Pull Request:

<!-- Include detailed instructions on how these changes can be tested. Review and follow the guide for how to write high-quality testing instructions. -->

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Enable the block_email_editor experimental flag (in DB, set woocommerce_feature_block_email_editor_enabled option to yes).
2. Navigate to WP Admin > Woo Emails
3. Create or edit an email template
4. Configure the email to be associated with a WC_Email using the select for email type in the right sidebar. This is a temporary way of setting the association.
![Screenshot 2025-03-07 at 12 45 08](https://github.com/user-attachments/assets/0662e84d-96d0-4f2d-b8ee-2f165361bcc8)

5. Do the action that triggers the email. For example, for `customer_processing_order` you need to create a new order.
6. You should receive the email created in the block editor and it should contain the main content from the WC_Email. Note: you can use [WP Mail Logging plugin](https://wordpress.org/plugins/wp-mail-logging/) to capture the emails

<!-- End testing instructions -->

### Testing that has already taken place:

<!-- Detail any testing that has already been conducted. -->
<!-- Include environment details such as hosting type, plugins, theme, store size, store age, and relevant settings. -->
<!-- Mention any analysis performed, such as assessing potential impacts on environment attributes and other plugins, performance profiling, or LLM/AI-based analysis. -->
<!-- Within the testing details you provide, please ensure that no sensitive information (such as API keys, passwords, user data, etc.) is included in this public pull request. -->

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box below and supplying data. -->
<!-- It will trigger the 'Add changelog to PR' CI job to create and push the entry into the branch. -->

<!-- Due to org permissions, the job might fail for PRs crated from a fork under GitHub organizations. Possible solutions: -->
<!-- * Create entry manually with `pnpm --filter='@woocommerce/plugin-woocommerce' changelog add` and push it into the branch (replace `@woocommerce/plugin-woocommerce` with package name from nearest `package.json` file) -->
<!-- * Create entry from supplied PR data and push it automatically `pnpm utils changefile pr-number-here -o github-org-name-here` -->

-   [ ] Automatically create a changelog entry from the details below.

<!-- If no changelog entry is required for this PR, you can specify that below and provide a comment explaining why. This cannot be used if you selected the option to automatically create a changelog entry above. -->

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

<!-- Choose only one -->

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message <!-- Add a changelog message here -->

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment <!-- If your Pull Request doesn't require a changelog entry, a comment explaining why is required instead -->

</details>

costasovo committed:
Alter styles for Woo emails when created via block editor

In the flow in WC_Email the content is generated and after styles are inlined.
We replace the email content when it is generated, but we need to provide styles
for inlining, which is done later in the flow.
We can apply only the block editor CSS or combine the editor's CSS with Woo email
CSS. I chose to combine because Woo email CSS contains some styles for the content which the editor's CSS doesn't cover.
I removed font-family and color styles from the Woo CSS to keep a unified look of the content.

costasovo committed:
Move the logic for capturing the Woo mail content to BlockEmailRenderer

This is the first step of refactoring the functionality for capturing
Woo email's content to an extra service.
We will need such a service to be able grab the content for block
or for preview.

costasovo committed:
Refactor functionality for capturing content from WC_Email into a service

The service extracts the main content from WC_Emails and inlines
Styles are generated from settings in the block editor.
The intention is to reuse it, for example, in the preview.

costasovo committed:
Add changelog

triple0t commented on plugins/woocommerce/src/Internal/EmailEditor/BlockEmailRenderer.php:
```diff
@@ -32,21 +32,31 @@ class BlockEmailRenderer {
-	final public function init(): void {
+	final public function init( WooContentProcessor $woo_content_processor ): void {
+		$this->woo_content_processor = $woo_content_processor;
 		add_action( 'woocommerce_blocks_renderer_initialized', array( $this, 'register_block_renderers' ) );
```
Missed this previously.
We need to rename this from `woocommerce_blocks_renderer_initialized` to `woocommerce_email_editor_blocks_renderer_initialized` to indicate it is an email editor core filter.
As this can be easily mistaken for a Woo core blocks filter

costasovo replied:
> Missed this previously. We need to rename this from `woocommerce_blocks_renderer_initialized` to `woocommerce_email_editor_blocks_renderer_initialized` to indicate it is an email editor core filter. As this can be easily mistaken for a Woo core blocks filter

@triple0t Good catch! As this is part of a renderer that theoretically can be used without the editor. What about renaming it to `woocommerce_email_blocks_renderer_initialized`? I would skip the `editor` part. Would you be ok with such a name?
I also see `woocommerce_blocks_renderer_parsed_blocks` so I would rename this one to `woocommerce_email_blocks_renderer_parsed_blocks`. Ok?

triple0t replied:
@costasovo that's okay with me.

costasovo replied:
Addressed in [c1482c2](https://github.com/woocommerce/woocommerce/pull/56199/commits/c1482c2d7dc04e7099278525a9c69815af9c90ad)

triple0t commented on plugins/woocommerce/src/Internal/EmailEditor/BlockEmailRenderer.php:
```diff
@@ -32,21 +32,31 @@ class BlockEmailRenderer {
+		$editor_container       = Email_Editor_Container::container();
+		$this->renderer         = $editor_container->get( EmailRenderer::class );
+		$this->personalizer     = $editor_container->get( Personalizer::class );
+		$this->theme_controller = $editor_container->get( Theme_Controller::class );
```
Left over?
![Screenshot 2025-03-10 at 12 47 58 PM](https://github.com/user-attachments/assets/64653848-f20a-4133-a6d9-1dd4564e1bb4)

Let's remove it if it is unused.

costasovo replied:
Addressed in [8eaa56d](https://github.com/woocommerce/woocommerce/pull/56199/commits/8eaa56d2f6869bab6727d3b76cce1ffc92283307).

triple0t reviewed (commented):
Well done. The code looks good to me, but I have some feedback.

The capture email does not have Text color applied when set with Global styles.

| Editor | Render |
| ------ | ----- |
|  ![Global-styles-editor](https://github.com/user-attachments/assets/94c2d8c8-158e-4ebb-ae05-0d39d2370b8a)  |     ![Global-styles-render](https://github.com/user-attachments/assets/7bb4650f-83ae-47a7-be40-811cd69613f1)  |

What worked

![Screenshot 2025-03-10 at 12 18 16 PM](https://github.com/user-attachments/assets/5a70a10b-f638-44c4-b918-157a390fdbc4)

* Font family (Block and Global Style) 
* Text colors (Block)
* Background color (Block and Global Style)

costasovo committed:
Remove unused service from BlockEmailRenderer

costasovo committed:
Update prefixes of block related filters and actions in email renderer

We need to distinguish general block filters from those related to email.
This commit changes prefix from `woocommerce_blocks` to `woocommerce_email_blocks`

costasovo commented:
@triple0t Thank you for the thorough review! I have addressed the remarks about an unused service and confusing filter prefixes.
I was not able to replicate the problem with the font color not being applied. I noticed on the screenshot that the color was not applied even on the paragraph block, which was not part of the change in this PR. So, I think it is possible that styles were not saved during the testing. Could you please try to replicate the issue and double-check that styles are saved after you change the text color?

![Screenshot 2025-03-17 at 14 22 13](https://github.com/user-attachments/assets/00a38576-acba-4c54-8f66-a41b46ba924d)

triple0t reviewed (approved):
LGTM. Well done.

Email Editor
![Screenshot 2025-03-17 at 5 22 30 PM](https://github.com/user-attachments/assets/0fe1d688-48b9-45b0-a7c8-36c551704045)

Preview
![Screenshot 2025-03-17 at 5 22 37 PM](https://github.com/user-attachments/assets/fff6aaf5-aeef-4256-9afe-5651323b3600)

Email Client
![Screenshot 2025-03-17 at 5 40 23 PM](https://github.com/user-attachments/assets/2beb3c85-c21c-47be-a5a6-2926353685b3)

I tested 
* Font family (Block and Global Style)
* Text colors (Block and Global Style)
* Background color (Block and Global Style)


Noting here: 

> > The capture email does not have Text color applied when set with Global styles.

> I was not able to replicate the problem with the font color not being applied. I noticed on the screenshot that the color was not applied even on the paragraph block, which was not part of the change in this PR. So, I think it is possible that styles were not saved during the testing. Could you please try to replicate the issue and double-check that styles are saved after you change the text color?

When I tested with the email logging plugin, the Global style text color settings were not applied, but on testing with an email client, the styles were applied correctly.
